Panic while parsing Pipenv lockfile

`osv-scanner@v1.2.0` (didn't test earlier versions) panics on `Pipfile.lock` files with `develop` packages without versions:
https://github.com/BenjMaq/bmaquet-tap-pagerduty/blob/cfa2fc8acb89fc6a30a701cbb29e13f0578436e6/Pipfile.lock


```
osv-scanner -L Pipfile.lock
panic: runtime error: slice bounds out of range [2:0]

goroutine 1 [running]:
github.com/google/osv-scanner/pkg/lockfile.ParsePipenvLock({0xc000034e80, 0x1c})
	~/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/pkg/lockfile/parse-pipenv-lock.go:58 +0x685
github.com/google/osv-scanner/pkg/lockfile.Parse({0xc000034e80, 0x1c}, {0xc0001f5de0, 0x0})
	~/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/pkg/lockfile/parse.go:141 +0x29f
github.com/google/osv-scanner/pkg/osvscanner.scanLockfile(0xc0001d8a80, 0xc000012cf0, {0xc000034e80, 0x1c}, {0xc0001f5de0, 0x0})
	~/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/pkg/osvscanner/osvscanner.go:191 +0x20d
github.com/google/osv-scanner/pkg/osvscanner.DoScan({{0xc0001cf940, 0x1, 0x1}, {0x0, 0x0, 0x0}, {0xf45650, 0x0, 0x0}, {0x0, ...}, ...}, ...)
	~/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/pkg/osvscanner/osvscanner.go:413 +0x2d1
main.run.func3(0xc0001d8500)
	~/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/cmd/osv-scanner/main.go:108 +0x45d
github.com/urfave/cli/v2.(*Command).Run(0xc0001906e0, 0xc0001d8500, {0xc000024180, 0x3, 0x3})
	~/go/pkg/mod/github.com/urfave/cli/v2@v2.24.4/command.go:273 +0xa42
github.com/urfave/cli/v2.(*App).RunContext(0xc000234000, {0xb76e90?, 0xc000026100}, {0xc000024180, 0x3, 0x3})
	~/go/pkg/mod/github.com/urfave/cli/v2@v2.24.4/app.go:332 +0x616
github.com/urfave/cli/v2.(*App).Run(...)
	~/go/pkg/mod/github.com/urfave/cli/v2@v2.24.4/app.go:309
main.run({0xc000024180, 0x3, 0x3}, {0xb72d60?, 0xc000014018}, {0xb72d60?, 0xc000014020})
	~/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/cmd/osv-scanner/main.go:127 +0x99e
main.main()
	~/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/cmd/osv-scanner/main.go:153 +0x46
```

--- a/pkg/lockfile/fixtures/pipenv/no-version.json
+++ b/pkg/lockfile/fixtures/pipenv/no-version.json
@@ -22,5 +22,11 @@
             "ref": "b36054111bc1e8bbadb5d0d60158feb72926f467"
         }
     },
-    "develop": {}
+    "develop": {
+        "itsdangerous": {
+            "editable": true,
+            "git": "https://github.com/pallets/itsdangerous",
+            "ref": "de09cad488a4d7c7bbcbcdb8e1c2dfde64325f48"
+        }
+    }
 }

--- a/pkg/lockfile/parse-pipenv-lock.go
+++ b/pkg/lockfile/parse-pipenv-lock.go
@@ -32,33 +32,27 @@ func ParsePipenvLock(pathToLockfile string) ([]PackageDetails, error) {
 		return []PackageDetails{}, fmt.Errorf("could not parse %s: %w", pathToLockfile, err)
 	}
 
-	packages := make(map[string]PackageDetails)
+	details := make(map[string]PackageDetails)
 
-	for name, pipenvPackage := range parsedLockfile.Packages {
+	addPkgDetails(details, parsedLockfile.Packages)
+	addPkgDetails(details, parsedLockfile.PackagesDev)
+
+	return pkgDetailsMapToSlice(details), nil
+}
+
+func addPkgDetails(details map[string]PackageDetails, packages map[string]PipenvPackage) {
+	for name, pipenvPackage := range packages {
 		if pipenvPackage.Version == "" {
 			continue
 		}
 
 		version := pipenvPackage.Version[2:]
 
-		packages[name+"@"+version] = PackageDetails{
+		details[name+"@"+version] = PackageDetails{
 			Name:      name,
 			Version:   version,
 			Ecosystem: PipenvEcosystem,
 			CompareAs: PipenvEcosystem,
 		}
 	}
-
-	for name, pipenvPackage := range parsedLockfile.PackagesDev {
-		version := pipenvPackage.Version[2:]
-
-		packages[name+"@"+version] = PackageDetails{
-			Name:      name,
-			Version:   version,
-			Ecosystem: PipenvEcosystem,
-			CompareAs: PipenvEcosystem,
-		}
-	}
-
-	return pkgDetailsMapToSlice(packages), nil
 }

